src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,838 @@
+
+        /// <summary>
+        /// Name of the product
+        /// </summary>
+        public const string Product = "AD7193";
+
+        /// <summary>
+        /// Category of the device
+        /// </summary>
+        public const string ProductCategory = "ADC";
+
+        /// <summary>
+        /// Description of the device
+        /// </summary>
+        public const string ProductDescription = "4-Channel, 4.8 kHz, Ultralow Noise, 24-Bit Sigma-Delta ADC with PGA";
+
+        /// <summary>
+        /// The URI of the datasheet of the device
+        /// </summary>
+        public const string DataSheetURI = "https://www.analog.com/media/en/technical-documentation/data-sheets/AD7193.pdf";
+        #endregion
+
+        #region Metadata for ISpiDevice
+
+        /// <summary>
+        /// The list of SPI modes that are valid for this device
+        /// </summary>
+        public const SpiMode ValidSpiModes = SpiMode.Mode3;
+
+        /// <summary>
+        /// The maximum frequency that can be used on the SPI bus
+        /// </summary>
+        public const int MaximumSpiFrequency = 10000000;
+        #endregion
+
+        #region metadata for IAdcDevice
+
+        /// <summary>
+        /// Number of ADCs on the device
+        /// </summary>
+        public const int ADCCount = 1;
+
+        /// <summary>
+        /// The maximum bitrate of the ADC
+        /// </summary>
+        public const int ADCBitrate = 24;
+
+        /// <summary>
+        /// The maximum sampling rate of the ADC
+        /// </summary>
+        public const int ADCSamplerate = 4800;
+
+        /// <summary>
+        /// The number of channels the ADC has
+        /// </summary>
+        public const int ADCInputChannelCount = 8;
+        #endregion
+
+        private readonly object _spiTransferLock = new object();
+        private readonly Stopwatch _stopWatch = new Stopwatch();
+
+        private SpiDevice _spiDevice = null;
+
+        /// <summary>
+        /// The list of received ADC values
+        /// </summary>
+        // protected ConcurrentQueue<AdcValue> adcValues = new ConcurrentQueue<AdcValue>();
+        protected BlockingCollection<AdcValue> adcValues = new BlockingCollection<AdcValue>();
+
+        /// <summary>
+        /// The event that is fired every time a new value is received from the ADC
+        /// </summary>
+        public event EventHandler<AdcValueReceivedEventArgs> AdcValueReceived;
+
+        private readonly StringBuilder _sb = new StringBuilder();
+
+        private readonly uint[] _registerCache = { 0x00, 0x080060, 0x000117, 0x000000, 0xa2, 0x00, 0x000000, 0x000000 };
+        private readonly byte[] _registerSize = { 1, 3, 3, 3, 1, 1, 3, 3 };
+
+        /// <summary>
+        /// The external reference voltage value. The default is 2.5V on REFIN1+ and REFIN1- (on the Digilent Pmod AD5 board)
+        /// </summary>
+        public double VReference { get; set; } = 2.50f;
```

Comment:
What is the use case of exposing this?

Reply:
The reference voltage can be set up in a way on the PCB, that is comes from an external source, connected to a particular pin on the IC. 
If the user decides to use an external voltage, they need to set this value manually to reflect their design.